rust/publish-ghost/src/lib.rs

```diff
@@ -33,6 +152,69 @@ pub struct Cli {
 
 impl Cli {
     pub async fn run(self) -> Result<()> {
-        Ok(())
+        if !self.path.exists() {
+            bail!("Path does not exist: {}", self.path.display())
+        }
+
+        if !self.path.is_file() {
+            bail!("Only publishing files is currently supported")
+        }
+
+        // Compile document
+        let doc = Document::open(&self.path).await?;
+        doc.compile(CommandWait::Yes).await?;
+
+        let theme = doc.config().await?.theme;
+        let node = &*doc.root_read().await;
+
+        // Convert it to Lexical
+        // TODO: use codec-lexical 
+
+        let options = EncodeOptions {
+            theme,
+            standalone: Some(false),
+            ..Default::default()
+        };
+
+        let codec = TextCodec;        
+        let (content, _encode_info) = codec.to_string(node, Some(options)).await?;
+        let content = format!("<!--kg-card-begin: html-->\r\n<pre>{content}</pre>\r\n<!--kg-card-end: html-->");
+
+        // Send content to Ghost
+        let token = generate_jwt(&self.key).context("generating JWT")?;
+
+        let url = if self.post {
+            format!("https://{}/ghost/api/admin/posts/", self.ghost)
+        } else if self.page {
+            format!("https://{}/ghost/api/admin/pages/", self.ghost)
+        } else {
+            unreachable!();
+        };
+
+        let payload = serde_json::json!({
+            "posts" : [
+                json!({
+                    "title": "WIP: Stencila",
+                    "html": content,
+                    "status": "published",
+                })
+            ]
+        });
+
+        let response = Client::new()
+            .post(url)
+            .header("Authorization", format!("Ghost {token}"))
+            .query(&[("source", "html")])
+            .json(&payload)
+            .send()
+            .await?;
+    
+        if response.status().is_success() {
+            Ok(())
```

Comment:
We will need to get the Ghost id of the page/post so we can add it to the metadata of the document.